component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthenticator.java

```diff
@@ -199,4 +209,117 @@ private List<String> populateMandatoryClaims() {
         mandatoryClaims.add(JWT_ID_CLAIM);
         return mandatoryClaims;
     }
+
+    /**
+     * Validate whether the request signing algorithm is registered for the application.
+     *
+     * @param request                 Http servlet request.
+     * @param bodyParameters          Map of request body params.
+     * @param oAuthClientAuthnContext OAuthClientAuthnContext.
+     * @return whether the request signing algorithm is registered for the application.
+     * @throws OAuthClientAuthnException
+     */
+    public boolean isRegisteredSignatureAlgorithm(HttpServletRequest request,
+                                                  Map<String, List> bodyParameters,
+                                                  OAuthClientAuthnContext oAuthClientAuthnContext)
+            throws OAuthClientAuthnException{
+
+        String signedObject = request.getParameter(OAUTH_JWT_ASSERTION);
```

Comment:
good to add some inline comments in this method logic.

Reply:
Fixed in c18c85cc171d19c53412ac6eaad0e1e4b738d909